src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.CheckOrReachability.cs

```diff
@@ -0,0 +1,1284 @@
+                        ImmutableArray<BoundPositionalSubpattern> newSubPatterns = discards.SetItem(i, deconstruction[i].WithPattern(newPattern));
+
+                        BoundPattern newRecursive = new BoundRecursivePattern(
+                            newPattern.Syntax, declaredType: node.DeclaredType, deconstructMethod: node.DeconstructMethod,
+                            deconstruction: newSubPatterns,
+                            properties: default, isExplicitNotNullTest: false, variable: null, variableAccess: null,
+                            node.InputType, node.NarrowedType, node.HasErrors);
+
+                        return saveMakeEvaluationSequenceOperand?.Invoke(newRecursive) ?? newRecursive;
+                    };
+
+                    for (; i < deconstruction.Length; i++)
+                    {
+                        VisitPatternAndCombine(node.Syntax, deconstruction[i].Pattern, startOfLeft);
+                    }
+
+                    _makeEvaluationSequenceOperand = saveMakeEvaluationSequenceOperand;
+                }
+
+                if (!node.Properties.IsDefault)
+                {
+                    var saveMakeEvaluationSequenceOperand = _makeEvaluationSequenceOperand;
+                    BoundPropertySubpattern? property = null;
+
+                    // Given `newPattern`, produce `{ Prop: newPattern }`
+                    _makeEvaluationSequenceOperand = (BoundPattern newPattern) =>
+                    {
+                        newPattern = WithInputTypeCheckIfNeeded(newPattern, property!.Pattern.InputType);
+                        ImmutableArray<BoundPropertySubpattern> newSubPatterns = [property.WithPattern(newPattern)];
+
+                        BoundPattern newRecursive = new BoundRecursivePattern(
+                            newPattern.Syntax, declaredType: node.DeclaredType, deconstructMethod: null, deconstruction: default,
+                            properties: newSubPatterns,
+                            isExplicitNotNullTest: false, variable: null, variableAccess: null,
+                            node.InputType, node.NarrowedType, node.HasErrors);
+
+                        return saveMakeEvaluationSequenceOperand?.Invoke(newRecursive) ?? newRecursive;
+                    };
+
+                    foreach (BoundPropertySubpattern subPattern in node.Properties)
+                    {
+                        property = subPattern;
+                        VisitPatternAndCombine(node.Syntax, property.Pattern, startOfLeft);
+                    }
+
+                    _makeEvaluationSequenceOperand = saveMakeEvaluationSequenceOperand;
+                }
+
+                _expectingOperandOfDisjunction = saveExpectingOperandOfDisjunction;
+
+                if (_evalSequence.Count - 1 < startOfLeft)
+                {
+                    // Everything was skipped
+                    TryPushOperand(node);
```

Comment:
>node [](http://example.com/codeflow?start=35&length=4)

This looks suspicious. What are we trying to accomplish by removing 
```
                    if (isEmptyPropertyPattern)
                    {
                        TryPushOperand(node);
                    }
```

above and making this change?

Are we assuming that ```isEmptyPropertyPattern``` is true? Why does it make sense to use the entire node otherwise? Also, is it appropriate to use the node at all if we are supposed to negate it?

Reply:
The thread was marked as resolved, but there was no any explanation given and the questions were not answered. I see changes made around this code, but that didn't clarify the intent and didn't lift correctness concerns.